src/mesh.cpp

```diff
@@ -1373,6 +1403,442 @@ bool RectilinearMesh::intersects(Position& r0, Position r1, int* ijk) const
+      for (int i=0; i<value.size(); i++)  coarse_z.push_back(value.at(i));
+    }
+
+    // Number of fine meshes within corresponding coarse meshes in z direction
+    if (check_for_node(node, "zints")) {
+      auto value = get_node_xarray<int>(node, "zints");
+      for (int i=0; i<value.size(); i++)  shape_z.push_back(value.at(i));
+      if (coarse_z.size() != shape_z.size())      fatal_error("The number of zmesh and zints must be same.");
+    }
+
+    // locations of fine meshes in z direction
+    for (int i=0; i<coarse_z.size(); i++) {
+      if (i==0) { 
+        width_z.push_back( (coarse_z.at(0)-lower_left_point[2])/shape_z.at(0) );
+        for (int j=0; j<shape_z.at(i); j++)  fine_z.push_back( lower_left_point[2]+width_z.back()*j );
+      } else {
+        width_z.push_back( (coarse_z.at(i)-coarse_z.at(i-1))/shape_z.at(i) );
+        for (int j=0; j<shape_z.at(i); j++)  fine_z.push_back( coarse_z.at(i-1)+width_z.back()*j );
+      }
+    }
+    fine_z.push_back(coarse_z.back());
+
+    mesh_ = std::make_unique<RectilinearMesh>(fine_x, fine_y, fine_z);
+    
+    // Energy group
+    if (check_for_node(node, "energy")) {
+      xml_node weightwindow_energy = node.child("energy");
+      // energy group for neutron
+      if (check_for_node(weightwindow_energy,"neutron")) {
+        n_ww=true; // turn on the flag
+        auto value = get_node_xarray<double>(weightwindow_energy, "neutron");
+        n_energy_group.push_back(0);
+        for (int i=0; i<value.size(); i++)  n_energy_group.push_back(value.at(i));
+      }
+             
+      // energy group for photon
+      if (check_for_node(weightwindow_energy,"photon")) {
+        if (!settings::photon_transport) { fatal_error("Photon transport is not on but weight window for photon is used"); }  // check if photon transport is on
+        p_ww=true; // turn on the flag
+        auto value = get_node_xarray<double>(weightwindow_energy, "photon");
+        p_energy_group.push_back(0);
+        for (int i=0; i<value.size(); i++)  p_energy_group.push_back(value.at(i));
+      }   
+    } else { fatal_error("Must assign energy group for weight window"); }  
+
+    // read wwinp file
+    if (check_for_node(node, "lower_ww")) {
+      auto value = get_node_xarray<double>(node, "lower_ww");
+      if ( value.size() != ( mesh_->n_bins()*( n_energy_group.size()+p_energy_group.size() ) ) ) 
+         { fatal_error("The number of lower weight window bounds is not the same as energy-space mesh cell numbers");  }
+      // neutron lower weight window bound first     
+      if (n_ww) {
+        for (int i=0; i<mesh_->n_bins()*n_energy_group.size(); i++) {   
+          n_ww_lower.push_back(value.at(i));
+        }  
+      } 
+    
+      // photon lower weight window bound later
+      if (p_ww) {
+        for (int i=0; i<mesh_->n_bins()*p_energy_group.size(); i++) {   
+          p_ww_lower.push_back(value.at(i+mesh_->n_bins()*n_energy_group.size()));
```

Comment:
I'd consider creating a variable for the offset value `mesh_->n_bins()*n_energy_group.size()` to make this line more clear.

Reply:
I created a new variable named `offset` which is equal to `mesh_->n_bins()*n_energy_group.size()`